### DISCUSSION TITLE
How to create Custom Layout in Theme Module

### DISCUSSION THREAD

**Discussion by hyzx86** [OrchardCMS/OrchardCore#12125] [Q&A] [answered]
Here is my testing , 
The follow image has describe my test infomation

My Theme module is  BasedOn  `TheAdmin` Theme.

If the code effect , the debug point will be stop , or the login page should display `33333333333333`

But it doesn't work~~

![image](https://user-images.githubusercontent.com/15613121/182038065-85e15e32-ddcc-429b-897e-a4b8ee089393.png)
![image](https://user-images.githubusercontent.com/15613121/182038107-c61541c0-ebec-4847-a3a1-d68f66f870c4.png)

**Comment by ns8482e** [comment, marked as answer]
@hyzx86 what makes you think that it should call `Layout-MyLogin.cshtml`. Did you have an alternated added for it? If not then it won't render  `33333333333333`

Rename `Layout-MyLogin.cshtml` to `Layout.cshtml` or add an alternate `Layout__MyLogin` to `Layout` shape

**Reply by hyzx86** [reply]
Thanks for your reply!

Sorry , I don't know how to do this yet. Is there any documentation in this regard? Where should I add it?

In addition, is there any way to override the `Layout_ Login` under the `TheAdmin`?

![image](https://user-images.githubusercontent.com/15613121/182039240-d90c15de-2c32-4259-9dbf-b85d2a83094a.png)

**Reply by hyzx86** [reply]
I was going to refer to `Layout__Login`'s registration method,

But I can't find where the `Layout__Login` is registered in the source code

**Reply by ns8482e** [reply]
To use admin theme for login uncheck `Use site theme for login page` In login settings

**Reply by hyzx86** [reply]
The `ViewLayout`  property will add an alternate, right?

https://github.com/OrchardCMS/OrchardCore/blob/339d74b251b53393dd510f8ad3027aa17d78500d/src/OrchardCore/OrchardCore.DisplayManagement/Razor/RazorPage.cs#L179

**Reply by ns8482e** [reply]
Yes it would, however you can only set it from other pages not from your Layout page

**Reply by hyzx86** [reply]
The cause of the problem is found. It's basetheme
I copied theadmintheme and renamed it to a new theme
Then from layout_ Login. Cshtml copies a layout_ VbenLogin.cshtml
Then in login Modify  Login.cshtml;

``` 
ViewLayout = "Layout_VbenLogin";
```

Then it's normal


![image](https://user-images.githubusercontent.com/15613121/182089710-89ad1815-ca36-42ce-9c4d-4885af10dc62.png)
